fix long codeblocks overflowing the div blog-post-content

s6muel opened the pull request:
Long lines of code within code blocks overflow the `blog-post-content` div, creating horizontal scroll bars. It does not occur with standard p, ul, etc.

Tested using Hugo 0.145.0, Firefox, and Chromium.

gevhaz commented:
Thanks for the PR, @s6muel. It's a valid issue that should be fixed.

Before your patch, it seems the code would overflow in that the text goes outside it's 'box', or the `blog-post-content` div as you say.

With your patch, we get soft-wrapping, so the line continues on the next line.

Since it's code, I think it's convention to not have soft line breaks. What do you think about adding a scroll bar instead? You mentioned something about it but I don't see one neither before nor after the patch. However, we can get one by adding `overflow: scroll` to the `pre` block near your existing change.

s6muel commented:
You're welcome mate.

Great point that I unfortunately overlooked in this PR. I've removed the `white-space: preserve` and added `overflow: auto` to the pre block. Auto seems to disregard the vertical scroll bars and only utilises horizontal scrolling when required--using scroll included vertical scrolling in Chromium and Firefox.

Cheers!

gevhaz commented:
Thanks for the thorough testing!

Merging.
